cluster-autoscaler/loop/trigger.go

```diff
@@ -43,38 +45,74 @@ type scalingTimesGetter interface {
 	LastScaleDownDeleteTime() time.Time
 }
 
+// provisioningRequestProcessingTimesGetter exposes recent provisioning request processing activity regardless of wether the
+// ProvisioningRequest was marked as accepted or failed. This is because a ProvisioningRequest being processed indicates that
+// there are other ProvisioningRequests that require processing regardless of the outcome of the current one. Thus, the next iteration
+// should be started immediately.
+type provisioningRequestProcessingTimesGetter interface {
+	LastProvisioningRequestProcessTime() time.Time
+}
+
 // LoopTrigger object implements criteria used to start new autoscaling iteration
 type LoopTrigger struct {
-	podObserver        *UnschedulablePodObserver
-	scanInterval       time.Duration
-	scalingTimesGetter scalingTimesGetter
+	initialized                          bool
+	podObserver                          *UnschedulablePodObserver
+	scanInterval                         time.Duration
+	scalingTimesGetter                   scalingTimesGetter
+	provisioningRequestProcessTimeGetter provisioningRequestProcessingTimesGetter
 }
 
 // NewLoopTrigger creates a LoopTrigger object
-func NewLoopTrigger(podObserver *UnschedulablePodObserver, scalingTimesGetter scalingTimesGetter, scanInterval time.Duration) *LoopTrigger {
+func NewLoopTrigger(scalingTimesGetter scalingTimesGetter, provisioningRequestProcessTimeGetter provisioningRequestProcessingTimesGetter, scanInterval time.Duration) *LoopTrigger {
 	return &LoopTrigger{
-		podObserver:        podObserver,
-		scanInterval:       scanInterval,
-		scalingTimesGetter: scalingTimesGetter,
+		podObserver:                          nil,
+		scanInterval:                         scanInterval,
+		scalingTimesGetter:                   scalingTimesGetter,
+		provisioningRequestProcessTimeGetter: provisioningRequestProcessTimeGetter,
+	}
+}
+
+// Initialize initializes the LoopTrigger object by providing a poiumer to the UnschedulablePodObserver
+func (t *LoopTrigger) Initialize(podObserver *UnschedulablePodObserver) {
+	t.podObserver = podObserver
+	t.initialized = true
+}
+
+// provisioningRequestWasProcessed is used to check if provisioningRequestProcessTimeGetter is not nil and a provisioning request was processed in the last iteration
+func (t *LoopTrigger) provisioningRequestWasProcessed(lastRun time.Time) bool {
```

Comment:
nit: on function ordering a good practice is to put public functions on the top and usually a function definition would be under its first usage. This way the order in the file starts from the highest level of abstraction and from most important functions in the package and you dig deeper into technical details as you go down the file.

Reply:
Updated